sapp/ui/frontend/src/Filter.js

```diff
@@ -788,7 +793,77 @@ const SaveFilterModal = (
     return <Alert type="error">{error.toString()}</Alert>;
   }
 
-  const onOk = (): void => {
+  let filterList = [];
+  let filterNamesList = [];
+
+  const isValidFilter = (filter: FilterDescription): Boolean => {
+    if (filter.name === undefined ||
+        filter.description === undefined ||
+        Object.keys(filter).length < 3
+    ) {
+      return false;
+    }
+    return true;
+  };
+
+  const importProps = {
+    multiple: true,
+    onRemove: filterFile => {
+      filterList.splice(filterList.indexOf(filterFile), 1);
+    },
+    beforeUpload(filterFile) {
+      return new Promise(resolve => {
+        if (filterFile.type !== 'application/json') {
+          message.error(`${filterFile.name} is not a json file`);
+          resolve(Upload.LIST_IGNORE);
+        } else {
+          let reader: FileReader = new FileReader();
+          reader.onload = (e) => {
+            let filter: FilterDescription = JSON.parse(e.target.result);
+            if (!isValidFilter(filter)) {
+              message.error(`${filterFile.name} is an invalid filter file`);
+              resolve(Upload.LIST_IGNORE);
+            } else {
+              filterList = [...filterList, filter];
+              filterNamesList = [...filterNamesList, `${filter.name}.json`]
+              resolve(filter);
+            }
+          };
+          reader.readAsText(filterFile);
+        }
+      });
+    },
+    customRequest: options => {
```

Comment:
```suggestion
    customRequest: {onSuccess, onError, file} => {
```

Reply:
Never knew this! Thanks for teaching ;)

Reply:
Are you sure about this? Maybe, I am missing something. But this doesn't work 🙃 Can you provide more information pleaseee?